apps/api/src/main/java/dk/treecreate/api/authentication/AuthController.java

```diff
@@ -115,4 +125,83 @@
             .authenticateUser(signUpRequest.getEmail(), signUpRequest.getPassword()));
     }
 
+    /**
+     * Refreshes the user's pair of tokens by generating a new set of <code>accessToken</code> and
+     * <code>refreshToken</code>, and invalidating the old pair.
+     * 
+     * @param request an HTTP request.
+     * @return a <code>JwtResponse</code> containing the new user information.
+     */
+    @GetMapping("/refresh")
+    public ResponseEntity<JwtResponse> refreshToken(
+        HttpServletRequest request) 
+    {
+        try
+        {
+            // Extract refresh token from request.
+            String token = jwtUtils.parseJwt(request);
+            String username = jwtUtils.getUserNameFromJwtToken(token);
+
+            UserDetailsImpl userDetails = (UserDetailsImpl) userDetailsService
+                .loadUserByUsername(username);
+            UsernamePasswordAuthenticationToken authentication =
+                new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            List<String> roles = userDetails.getAuthorities().stream()
+                .map(GrantedAuthority::getAuthority)
+                .collect(Collectors.toList());
+
+            // Remove the old pair of tokens from the whitelist.
+            jwtUtils.removeWhitelistJwtPair(token); 
+
+            // Generate a new set of tokens for the user.
+            String accessToken = jwtUtils.generateJwtToken(authentication);
+            String refreshToken = jwtUtils.generateJwtRefreshToken(authentication);
+
+            // Whitelist the new pair of tokens.
+            jwtUtils.whitelistJwtPair(accessToken, refreshToken);
+            
+            return ResponseEntity.ok(new JwtResponse(accessToken, 
+                refreshToken,
+                userDetails.getUsedId(),
+                userDetails.getEmail(),
+                userDetails.getIsVerified(),
+                roles));
+        } catch (Exception e)
+        {
+            LOGGER.error("Failed to refresh the authentication token.", e);
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
+                "Failed to refresh the authentication token. Try again later.");
+        }   
+    }
+
+    /**
+     * Logs out the user by taking the token from the <code>Authorization</code> header, 
+     * extracting the username from its body, and invalidating all of the tokens that 
+     * belong to that user.
+     * 
+     * @param request an HTTP request.
+     * @return a <code>Response Entity</code> with no body.
+     */
+    @GetMapping("/logout")
```

Comment:
Should specify that it returns HTTP No Content status
```suggestion
    @GetMapping("/logout")
    @ResponseStatus(HttpStatus.NO_CONTENT)
```